agent/app/agent.go

```diff
@@ -330,6 +331,7 @@ func (agent *ecsAgent) doStart(containerChangeEventStream *eventstream.EventStre
 		agent.saveMetadata(data.ClusterNameKey, agent.cfg.Cluster)
 		agent.saveMetadata(data.ContainerInstanceARNKey, agent.containerInstanceARN)
 		agent.saveMetadata(data.EC2InstanceIDKey, currentEC2InstanceID)
+		agent.saveMetadata(data.RegistrationTokenKey, agent.registrationToken)
```

Comment:
this is after RCI. should we save this before calling RCI? so that if the agent stops before RCI finishes, it can call with the same token when starting up again